File: solutions/71.simplify-path/src/stack.ts
/*
 * @lc app=leetcode.cn id=71 lang=typescript
 *
 * https://leetcode.cn/problems/simplify-path/
 */

// @lc code=start
function simplifyPath(path: string): string {
    const stack: string[] = [];
    for (const part of path.split('/')) {
        if (part === '..') {
            stack.pop();
        } else if (part !== '.' && part !== '') {
            stack.push(part);
        }
    }
    return '/' + stack.join('/');
}
// @lc code=end

export { simplifyPath };

File: solutions/71.simplify-path/src/stack.rs
/*
 * @lc app=leetcode.cn id=71 lang=rust
 *
 * https://leetcode.cn/problems/simplify-path/
 */

pub struct Solution;

// @lc code=start
impl Solution {
    pub fn simplify_path(path: String) -> String {
        let mut stack = vec![];
        for p in path.split("/") {
            match p {
                "" | "." => {}
                ".." => {
                    stack.pop();
                }
                _ => {
                    stack.push(p);
                }
            }
        }
        "/".to_string() + &stack.join("/")
    }
}
// @lc code=end
